### PR TITLE
Refactor cluster data flow

### PR DESCRIPTION
**Pull request by ograu**
Towards https://github.com/giantswarm/giantswarm/issues/7521

Sorry for the size of the PR, I have had to do a lot of changes...

In this PR:

- Added reducer for loading flags: the idea is to start removing loading flags from all other reducers and have this 'no-maintenance' reducer that will take care of all the flags as long as we follow the convention of adding `request/success/error/finished` suffixes to actions. We can use `finished` when we don't want to use `success/error`. There are some situations when errors are handled by other functions and we just want to know when the call is finished.

- Added request/success/error/finished action types

- Added options object to some action creators for setting/not setting loading flags

- Changed tests _without nodes/nps ready_ to actually check for 0 nodes and leave as todos with `.skip`s the tests with nodes/node pools

- Changed `isNodePoolsCluster` for `isV5Cluster`

- Removed the use of `.then()` in components to avoid memory leaks. And also as a good practice, because in most of the cases we want to use redux store instead of local state.

### CODE REVIEW

File: src/components/Home/Home.js [hunk 1]
```diff
@@ -57,9 +59,13 @@ class Home extends React.Component {
    */
   title() {
     if (this.props.selectedOrganization) {
-      return `Cluster Overview | ${this.props.selectedOrganization}`;
+      return (
+        'Cluster Overview | ' +
+        this.props.selectedOrganization +
+        ' | Giant Swarm'
```
Comment (axbarsan) [inline src/components/Home/Home.js]: The `| Giant Swarm` suffix gets automatically added by the `useDocumentTitle` hook inside the `DocumentTitle` component

**Comment by axbarsan** [comment]
Really like the universal loading reducer idea. 👍 
However, I think it should be heavily documented, as there's no obvious clue to its usage

**Comment by ograu** [comment]
Thanks @axbarsan !

Regarding documentation, totally agree with you. Still thinking on how and where to write documentation for this. Any suggestion will be appreciated.

File: __tests__/V4AWSClusterManagement.js [hunk 2]
```diff
@@ -86,10 +86,10 @@ it('renders all the v4 AWS cluster data correctly without nodes ready', async ()
   expect(getAllByText(V4_CLUSTER.id)).toHaveLength(2);
   expect(
     getByTestId('desired-nodes').querySelector('div:nth-child(2)').textContent
-  ).toBe('3');
+  ).toBe('0');
   expect(
     getByTestId('running-nodes').querySelector('div:nth-child(2)').textContent
-  ).toBe('3');
+  ).toBe('0');
```
Comment (oponder) [inline __tests__/V4AWSClusterManagement.js]: why did this become 0?

File: __tests__/V4KVMClusterManagement.js [hunk 3]
```diff
@@ -91,12 +91,20 @@ it('renders all the v4 KVM cluster data correctly without nodes ready', async ()
...
-    status: v4KVMClusterStatusResponse,
-  });
-  expect(nodes).toHaveTextContent(nodesRunning);
+  const nodes = getByText('Nodes').nextSibling.textContent;
+  expect(nodes).toBe('0');
+});
+
+/******************** PENDING TESTS ********************/
+
+it.skip('renders all the v4 KVM nodes', async () => {
```
Comment (oponder) [inline __tests__/V4KVMClusterManagement.js]: I guess there are some changes in the tests still to do?

File: src/components/Layout.js [hunk 4]
```diff
@@ -54,7 +54,7 @@ class Layout extends React.Component {
   render() {
     return (
       <DocumentTitle>
-        <LoadingOverlay loading={!this.props.firstLoadComplete}>
+        <LoadingOverlay loading={this.props.loadingClustersList !== false}>
```
Comment (oponder) [inline src/components/Layout.js]: Can this be `this.props.loadingClustersList` instead of `this.props.loadingClustersList !== false` ?

Comment (oponder) [inline src/components/Layout.js]:
```suggestion
        <LoadingOverlay loading={this.props.loadingClustersList}>
```

File: src/components/Organizations/Organizations.js [hunk 5]
```diff
@@ -11,11 +12,13 @@ const Organizations = props => {
 
   return (
     <Breadcrumb data={{ title: 'ORGANIZATIONS', pathname: url }}>
-      <Switch>
-        <Route component={List} exact path={`${path}`} />
-        <Route component={Detail} path={`${path}/:orgId`} />
-        <Redirect path={`${path}*`} to={`${url}`} />
-      </Switch>
+      <LoadingOverlay loading={props.loadingClustersDetails !== false}>
```
Comment (oponder) [inline src/components/Organizations/Organizations.js]: Same suggestion. Unless I am missing something or having a boolean logic brainfart.

```suggestion
      <LoadingOverlay loading={props.loadingClustersDetails}>
```

**Comment by oponder** [comment]
Thanks for the effort and hard work here so far!

File: __tests__/V4AWSClusterManagement.js (hunk 2 above)
Comment (ograu) [reply __tests__/V4AWSClusterManagement.js]: Same question I am asking to myself... Fixed for all v4 tests. Thanks!

File: __tests__/V4KVMClusterManagement.js (hunk 3 above)
Comment (ograu) [reply __tests__/V4KVMClusterManagement.js]: Test removed

File: src/components/Layout.js (hunk 4 above)
Comment (ograu) [reply src/components/Layout.js]: Changed: the only downside is that now flags have to be initialized to avoid undefined flags.

Comment (oponder) [reply src/components/Layout.js]: Oh I'm ok if this is done with a reason. It just read a bit weird to me.

**Pull request merged**